lib/board.rb

```diff
@@ -0,0 +1,30 @@
+class Board
+  DENOTE_EMPTY = '-'
+  attr_reader :board, :num_of_blank_spaces, :blank_positions
+
+  def initialize(size)
+    @board = Array.new(size) { Array.new(size) { DENOTE_EMPTY } }
+    @num_of_blank_spaces = size * size
+    @blank_positions = make_free_positions_hash(size.times.to_a.product(size.times.to_a))
+  end
+
+  def print_board(output = $stdout)
+    board.each do |i|
+      output.puts i.to_s
+    end
```

Comment:
This could potentially go to user interface module.

Reply:
Resolved at 63df55f75eff909dfefea0e00207b6d6d96703a2